src/components/TimeDurationControl/TimeBox.tsx

```diff
@@ -87,23 +87,29 @@ export const TimeBox: FC<TimerProps> = ({
   const handleChangeInput = (e: React.FormEvent<HTMLInputElement>) => {
     let input = e.currentTarget.value;
 
-    input = input.replace(/[^0-9]+/g, '');
-
-    input = `${input.substr(0, 2)}:${input.substr(2, 2)}:${input.substr(4, input.length)}`;
-
-    input = input.substring(0, 9);
+    input = input.replace(/\D/g, '')
```

Comment:
Currently this does not allow for a user to backspace a value, and type in the new value of some portion. It is really hard to accurately input the times you need to. I think we need some improvements to the user experience here.